cpp/include/cudf/fixed_point/fixed_point.hpp

```diff
@@ -31,20 +31,8 @@
 
 //! `fixed_point` and supporting types
 namespace numeric {
-/** \cond HIDDEN_SYMBOLS */
-// This is a wrapper struct that enforces "strong typing"
-// at the construction site of the type. No implicit
-// conversions will be allowed and you will need to use the
-// name of the type alias (i.e. scale_type{0})
-template <typename T>
-struct strong_typedef {
-  T _t;
-  CUDA_HOST_DEVICE_CALLABLE explicit constexpr strong_typedef(T t) : _t(t) {}
-  CUDA_HOST_DEVICE_CALLABLE operator T() const { return _t; }
-};
-/** \endcond */
 
-using scale_type = strong_typedef<int32_t>;
+enum scale_type : int32_t {};
```

Comment:
```suggestion
enum class scale_type : int32_t {};
```

Does this enum contain any enum value?

Reply:
For our purposes, we want implicit conversions. What we want is enforcing the "strong typing" at the call site. See here: https://www.godbolt.org/z/b38xdTqoc. As for the second question, this is just the pattern of how to use direct-list-initialiation for "strong typing".

Reply:
> For our purposes, we want implicit conversions. 

So a scale_type variable has to be explicitly typed to create, but it implicitly casts to an int. Is this correct?

Reply:
Yep!